scipy/spatial/transform/rotation.py

```diff
@@ -657,3 +657,119 @@ def __mul__(self, other):
         if self._single and other._single:
             result = result[0]
         return self.__class__(result, normalized=True)
+
+    def __getitem__(self, val):
+        """Extract rotation at given index(es) from object.
+
+        This function returns a new `Rotation` instance containing:
+
+            - a single rotation, if `val` is a single index
+            - a stack of rotation(s), if `val` is a slice, or an index array.
+              In cases where a single index is ultimately specified, the stack
+              will contain a single rotation.
+
+        Parameters
+        ----------
+        val : index, slice, or index array
+            This parameter specifies which rotation(s) to extract.
+        """
+        # __init__ now copies for all cases
+        return self.__class__(self._quat[val], normalized=True)
+
+    def copy(self):
```

Comment:
Do we need this function after you added copying in the constructor?

Reply:
No we don't need it for our code. Should I leave it as a function for the user or remove it?

Reply:
Considering our object is now essentially immutable, I don't see it being necessary at all.